packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -213,7 +213,21 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
         return;
       }
 
-      const newParentObject = getObject3DBySceneNodeRef(newParentRef);
+      let newParentObject = getObject3DBySceneNodeRef(newParentRef);
+      const newParent = getSceneNodeByRef(newParentRef);
+
+      // If the direct new parent is a sub model node, use its modelRef parent
+      // to calculate the new transforms to prevent the object froming being
+      // moved to random position.
+      if (findComponentByType(newParent, KnownComponentType.SubModelRef)) {
+        let hierarchyParent = newParent;
```

Comment:
No need of 'hierarchyParent'. newParent can be changed from const to let and re-used here.

Reply:
Good catch!

Reply:
they don't have the same meaning. the `newParent` is the scene node that the node will be reparented to. it shouldn't be changed, no matter it's a submodel node or not, so if the later code uses it, it will be the correct node. the `hierarchyParent` is a local variable that's only used for find the hierarchy parent object. I guess I can rename `newParentObject` to `newHierarchyParentObject` to make it unrelated to `newParent`

Reply:
Understood. Makes sense now.